### PR TITLE
UGENE-7237. Change FASTA format to write 80-characters per line

### PR DESCRIPTION
**Pull request by yalgaer**

### CODE REVIEW

File: src/corelibs/U2Formats/src/FastaFormat.cpp [hunk 1]
```diff
@@ -317,19 +317,24 @@ static void writeHeaderToFile(IOAdapterWriter &writer, const QString &sequenceNa
...
+        QByteArray chunkContent = sequence->getSequenceData(region, os);
         CHECK_OP(os, );
         writer.write(os, QString::fromLatin1(chunkContent));
         CHECK_OP(os, );
+        if (region.endPos() < sequenceLength) {
+            writer.write(os, "\n");
+            CHECK_OP(os, );
+        }
     }
     writer.write(os, "\n");
```
Comment (DmitriiSukhomlinov) [inline src/corelibs/U2Formats/src/FastaFormat.cpp:339]: Now you have two endline characters in the end of document. I think, we can remove this line

File: src/corelibs/U2Formats/src/FastaFormat.cpp [hunk 2]
```diff
@@ -317,19 +317,24 @@ static void writeHeaderToFile(IOAdapterWriter &writer, const QString &sequenceNa
...
-        QByteArray chunkContent = sequence->getSequenceData(U2Region(i, chunkSize), os);
+    for (int i = 0; i < sequenceLength; i += FASTA_SEQUENCE_LINE_LENGTH) {
+        qint64 chunkSize = qMin(FASTA_SEQUENCE_LINE_LENGTH, sequenceLength - i);
+        U2Region region(i, chunkSize);
+        QByteArray chunkContent = sequence->getSequenceData(region, os);
         CHECK_OP(os, );
         writer.write(os, QString::fromLatin1(chunkContent));
         CHECK_OP(os, );
+        if (region.endPos() < sequenceLength) {
+            writer.write(os, "\n");
```
Comment (DmitriiSukhomlinov) [inline src/corelibs/U2Formats/src/FastaFormat.cpp]: I think, you can text `\n` here every time and remove the same line below

File: src/corelibs/U2Formats/src/FastaFormat.cpp [hunk 3]
```diff
@@ -339,11 +344,15 @@ static void saveSequence(IOAdapterWriter &writer, const DNASequence &sequence, U
...
-    qint64 len = sequence.length();
-    for (qint64 i = 0; i < len; i += SAVE_LINE_LEN) {
-        int chunkSize = (int)qMin(SAVE_LINE_LEN, len - i);
+    qint64 sequenceLength = sequence.length();
+    for (qint64 i = 0; i < sequenceLength; i += FASTA_SEQUENCE_LINE_LENGTH) {
+        int chunkSize = (int)qMin(FASTA_SEQUENCE_LINE_LENGTH, sequenceLength - i);
         writer.write(os, QString::fromLatin1(seq + i, chunkSize));
         CHECK_OP(os, );
+        if (i + chunkSize < sequenceLength) {
+            writer.write(os, "\n");
```
Comment (DmitriiSukhomlinov) [inline src/corelibs/U2Formats/src/FastaFormat.cpp]: The same

File: src/corelibs/U2Formats/src/FastaFormat.cpp (hunk 2 above)
Comment (yalgaer) [reply src/corelibs/U2Formats/src/FastaFormat.cpp]: Fixed.

File: src/corelibs/U2Formats/src/FastaFormat.cpp (hunk 3 above)
Comment (yalgaer) [reply src/corelibs/U2Formats/src/FastaFormat.cpp]: Fixed

**Pull request merged**